crates/pathfinder/src/bin/pathfinder/config.rs

```diff
@@ -247,13 +247,14 @@ This should only be enabled for debugging purposes as it adds substantial proces
     get_events_max_uncached_bloom_filters_to_load: std::num::NonZeroUsize,
 
     #[arg(
-        long = "storage.prune-state-tries",
-        long_help = "When enabled, only the last N+1 states of the Merkle tries are kept in the database. \
+        long = "storage.state-tries",
+        long_help = "When set to `archive` all historical Merkle trie state is preserved. When set to an integer N, only the last N+1 states of the Merkle tries are kept in the database. \
             This can be used to reduce the disk space usage at the cost of only being able to provide storage proofs for the latest N+1 blocks (the state for the latest block is always stored).",
-        env = "PATHFINDER_PRUNE_MERKLE_TRIES",
-        value_name = "N"
+        env = "PATHFINDER_STORAGE_STATE_TRIES",
+        value_name = "archive | N",
+        value_parser = parse_state_tries
     )]
-    prune_merkle_tries: Option<u64>,
+    state_tries: Option<StateTries>,
```

Comment:
I think we should mention the default behavior and value here. If so, then the Option can be omitted because we could use clap's default mechanism.

Reply:
Yeah, I did consider that. The reason I did not mention it is that the default value is quite complicated... It depends on both `is_new_database` and if the prune flag is set in the database. That's the reason why we can't just simply specify a default value here either.

There's also a difference in behavior when not specifying this CLI argument at all (defaults to a sane value depending on DB state) vs explicitly requesting something.

Reply:
But we could definitely describe this behavior in the help message. Would that be reasonable?

Reply:
> The reason I did not mention it is that the default value is quite complicated... It depends on both is_new_database and if the prune flag is set in the database.

Personally I think we should just enforce a default value of some `N` value. Existing nodes must then set `archive` but I think that's fine. This way its explicit and the default behaviour is simpler.

Reply:
> Existing nodes must then set archive but I think that's fine.

That's something I wanted to avoid. The only possible choice for existing nodes is archive mode so why force them to add more configuration / command line arguments just to keep things working?

Reply:
To prevent confusion about what mode they're running in and to reduce complexity.

My view is that if we were starting new today; they should opt-into this behaviour and not just have it happen.

Its a breaking version; they can add a new configuration argument?

Reply:
> To prevent confusion about what mode they're running in and to reduce complexity.

If you're upgrading that should be quite clear even without forcing an extra command line argument on you.

> My view is that if we were starting new today; they should opt-into this behaviour and not just have it happen.

You mean `archive`? That's true with this change, too. When starting new you get pruning enabled by default.

> Its a breaking version; they can add a new configuration argument?

I don't really see a reason given we _do_ have a way to just make it work seamlessly? Why make it harder for users if we can do better?

Reply:
I think having diverging behaviour for lack-of-setting a value isn't great - its confusing for everyone involved. And I think the vast majority of users will elect to "resync" or use the new snapshot, so we're really only keeping this behaviour for a handful of users.

But alright; its not the biggest problem, lets see how it goes